### ISSUE TITLE
Transitive dependencies of a intellijPlatformPluginModule are lost

### ISSUE THREAD

**Issue by abrooksv** [JetBrains/intellij-platform-gradle-plugin#1655]
### What happened?

Using `intellijPlatformPluginModule` on a module that brings in 3rd party libs causes the lips to be lost when building the final zip

### Steps to reproduce

* Have moduleA depend on a 3rd party jar such as Commonmark
* Have top level project use `intellijPlatformPluginModule(project(":moduleA"))`
* Run `:buildPlugin`, see Zip contain the merged moduleA's jar into the composedJar, but the zip is missing the commonmark jar as a sibling jar (since composeJar isn't supposed to do any shadowing on 3rd party libs)

Doing the following works-ish by bringing in the transitive deps:
```
    implementation(project(":moduleA")) {
        intellijPlatformPluginModule(this)
    }
```

but it composes the jar from `moduleA` and includes the `moduleA-base.jar` into the Zip which seems incorrect as well

### Gradle IntelliJ Plugin version

2.0.0-beta7

### Gradle version

8.8

### Operating System

None

**Comment by hsz** [comment]
@abrooksv could you please verify it once again with `2.0.0-beta9`?

**Comment by abrooksv** [comment]
Yes, it is still broken

**Comment by hsz** [comment]
I have just verified that on my local, and it seems to be working just fine.

The folowing is required to be present in the root module:

```kotlin
dependencies {
    intellijPlatform {
        implementation(project(":module")) {
            pluginModule(this)
        }
    }
}
```

The `implementation(project(":module"))` part is adding your module to the `implementation` configuration, and with `pluginModule` (AKA `intellijPlatformPluginModule`) you just **mark** this dependency as to-be-bundled within the main Jar.

With that, the `:module` Jar is consumed into the `:` Jar archive, and all `:module` transitive dependencies are present in the `lib` directory.
The root jar is also properly composed with the final jar, not the `-base` one.

**Comment by abrooksv** [comment]
I hacked your test to show it being broken:

```
Subject: [PATCH] Example of broken
---
Index: src/integrationTest/resources/multi-module/submodule/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/resources/multi-module/submodule/build.gradle.kts b/src/integrationTest/resources/multi-module/submodule/build.gradle.kts
--- a/src/integrationTest/resources/multi-module/submodule/build.gradle.kts	(revision 63cd4ed561e78735d0413200f7139bbc71014cf0)
+++ b/src/integrationTest/resources/multi-module/submodule/build.gradle.kts	(date 1720820382923)
@@ -28,3 +28,7 @@
         instrumentationTools()
     }
 }
+
+intellijPlatform {
+    instrumentCode = false
+}
\ No newline at end of file
Index: src/integrationTest/resources/multi-module/base/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/resources/multi-module/base/build.gradle.kts b/src/integrationTest/resources/multi-module/base/build.gradle.kts
--- a/src/integrationTest/resources/multi-module/base/build.gradle.kts	(revision 63cd4ed561e78735d0413200f7139bbc71014cf0)
+++ b/src/integrationTest/resources/multi-module/base/build.gradle.kts	(date 1720822114919)
@@ -37,4 +37,5 @@
 
 intellijPlatform {
     buildSearchableOptions = false
+    instrumentCode = false
 }
Index: src/integrationTest/resources/multi-module/ext/build.gradle.kts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/resources/multi-module/ext/build.gradle.kts b/src/integrationTest/resources/multi-module/ext/build.gradle.kts
--- a/src/integrationTest/resources/multi-module/ext/build.gradle.kts	(revision 63cd4ed561e78735d0413200f7139bbc71014cf0)
+++ b/src/integrationTest/resources/multi-module/ext/build.gradle.kts	(date 1720822124273)
@@ -27,8 +27,10 @@
         instrumentationTools()
 
         localPlugin(project(":base"))
-        pluginModule(implementation(project(":submodule")))
+//        pluginModule(implementation(project(":submodule")))
     }
+
+    intellijPlatformPluginModule(project(":submodule"))
 
     implementation(project(":raw"))
     implementation("org.jetbrains:markdown:0.7.3")
@@ -36,4 +38,5 @@
 
 intellijPlatform {
     buildSearchableOptions = false
+    instrumentCode = false
 }
Index: src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/MultiModuleIntegrationTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/MultiModuleIntegrationTest.kt b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/MultiModuleIntegrationTest.kt
--- a/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/MultiModuleIntegrationTest.kt	(revision 63cd4ed561e78735d0413200f7139bbc71014cf0)
+++ b/src/integrationTest/kotlin/org/jetbrains/intellij/platform/gradle/MultiModuleIntegrationTest.kt	(date 1720821902097)
@@ -3,6 +3,8 @@
 package org.jetbrains.intellij.platform.gradle
 
 import org.jetbrains.intellij.platform.gradle.Constants.Tasks
+import java.nio.file.FileSystems
+import kotlin.io.path.listDirectoryEntries
 import kotlin.test.Test
 
 class MultiModuleIntegrationTest : IntelliJPlatformIntegrationTestBase(
@@ -11,8 +13,18 @@
 
     @Test
     fun `ext plugin uses base plugin from zip distribution`() {
-        build(":ext:" + Tasks.BUILD_PLUGIN, projectProperties = defaultProjectProperties) {
-
+        build(
+            gradleVersion = gradleVersion,
+            fail = false,
+            assertValidConfigurationCache = false,
+            ":ext:" + Tasks.BUILD_PLUGIN,
+            projectProperties = defaultProjectProperties,
+        ) {
+            val pluginZip = dir.resolve("ext/build/distributions/ext.zip")
+            FileSystems.newFileSystem(pluginZip, emptyMap<String, Any>()).use { zipFile ->
+                val files = zipFile.getPath("ext/lib").listDirectoryEntries()
+                println("Files: $files")
+            }
         }
     }
 }
```

TL;DR:
` pluginModule(implementation(project(":submodule")))` works
`intellijPlatformPluginModule(project(":submodule"))` is broken

**Comment by hsz** [comment]
> `intellijPlatformPluginModule(project(":submodule"))` is broken

It's expected to be broken as you're not supposed to use it alone without `implementation`!

**Comment by abrooksv** [comment]
Can I go hide in shame now?

**Comment by hsz** [comment]
The `intellijPlatformPluginModule()` configuration (or its alias: `intellijPlatform { pluginModule() }`) is just a marker to highlight what dependencies should be consumed into the main jar (without their transitive dependencies).
You still need to add such dependencies to the `implementation()` configuration to make them present in the actual compilation classpath.

**Comment by abrooksv** [comment]
Okay, so how should I model Root->Sub Module2->Sub Module 1?

I am getting duplicate jars in my zip :(

```
submodule2-base.jar
submodule1-base.jar
plugin.jar
```

**Comment by abrooksv** [comment]
Ok, I think we can close this then
